flask_hal/document.py

```diff
@@ -82,3 +95,65 @@ def to_json(self):
         """
 
         return json.dumps(self.to_dict())
+
+
+class Document(BaseDocument):
+    """Constructs a ``HAL`` document.
+    """
+
+    def __init__(self, data=None, links=None, embedded=None):
+        """Initialises a new ``HAL`` Document instance. If no arguments are
+        provided a minimal viable ``HAL`` Document is created.
+
+        Keyword Args:
+            data (dict): Data for the document
+            links (flask_hal.link.Collection): A collection of ``HAL`` links
+            embedded: TBC
+
+        Raises:
+            TypeError: If ``links`` is not a :class:`flask_hal.link.Collection`
+        """
+        super(Document, self).__init__(data, links, embedded)
+        self.links.append(link.Self())
+
+
+class Embedded(BaseDocument):
+    """Constructs a ``HAL`` embedded.
+
+    Example:
+        >>> document = Document(
+        >>>     embedded={
+        >>>         'orders': Embedded(
+        >>>             embedded={'details': Embedded(
+        >>>                 data={'details': {}}
+        >>>             )},
+        >>>             links=link.Collection(
+        >>>                 link.Link('foo', 'www.foo.com'),
+        >>>                 link.Link('boo', 'www.boo.com')
+        >>>             ),
+        >>>             data={'total': 30},
+        >>>         )
+        >>>     },
+        >>>     data={'currentlyProcessing': 14}
+        >>> )
+        >>> document.to_json()
+        ... {
+                "_links": {
+                    "self": {"href": "http://localhost/entity/231"}
+                },
+                "_embedded": {
+                    "orders": {
+                        "_embedded": {
```

Comment:
Sorry don't get it.. Links are on line 150. What do you means? And `_links` are options so when it's empty document doesn't return them.